include/ginkgo/core/base/executor.hpp

```diff
@@ -1355,26 +1307,38 @@ class EnableDeviceReset {
      *
      * @param device_reset  whether to allow a device reset or not
      */
-    void set_device_reset(bool device_reset) { device_reset_ = device_reset; }
+    [[deprecated(
+        "device_reset is no longer supported, call "
+        "cudaDeviceReset/hipDeviceReset manually")]] void
```

Comment:
what do you have in mind? we have wrappers for the functions internally already

Reply:
On the question whether it needs to be used, I feel like most of the guidance is pretty old - I just checked again, the requirement to call cudaDeviceReset for `compute-sanitizer` has been removed in a recent version, the NSight Systems documentation is a bit unclear, especially what happens in the case of crashes?

> Nsight Systems will not have information about CUDA events that were still in device buffers when analysis terminated. It is a good idea, if using cudaProfilerAPI to control analysis to call cudaDeviceReset before ending analysis.

>  Normally, in order to collect full CUDA trace, the application needs to finalize the device used for CUDA work (call cudaDeviceReset(), and then let the application gracefully exit (as opposed to crashing).

Your suggestion may lead to some issues, since it requires the executor to exist until the `reset_device` function is called, but by calling it we basically destroy all allocations, so ideally we should be calling this outside the scope of the executor, like
```
int main() {
    {
        run_application();
    }
    cudaDeviceReset();
}
```

Reply:
It seems with both `ncu` and `nsys`, this branch (without the device reset) works the same as develop, so maybe they dont require a device_reset anymore ? 

With that, I guess most applications should not have a need to call reset anymore, so I guess it is okay that we dont provide that anymore.